### PR TITLE
docs: add a hello-craft-parts tutorial

### PR DESCRIPTION
**Pull request by lengau**
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

I wrote this as a way to help myself understand what I needed for my actual craft-parts sprint work, so if we don't want it that's fine. I just figured since I'd created it I might as well push it up.

I know craft-parts doesn't (yet?) have spread tests, but I used the rockcraft docs as my template and set it up to be ready for spread testing

### CODE REVIEW

**Comment by lengau** [comment]
Linting issues are fixed in #359, will rebase once that's merged.

File: docs/tutorials/code/hello-craft-parts/parts-empty.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+parts:
+  hello:
+    plugin: nil
```
Comment (lengau) [inline docs/tutorials/code/hello-craft-parts/parts-empty.yaml:3]:
```suggestion
    plugin: nil

```

File: docs/tutorials/code/hello-craft-parts/parts-pull.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+parts:
+  hello:
+    plugin: nil
+    source: hello.sh
+    source-type: file
```
Comment (lengau) [inline docs/tutorials/code/hello-craft-parts/parts-pull.yaml:5]:
```suggestion
    source-type: file

```

**Review by tigarmo — APPROVED** [review]
very nice, thanks for the initiative!

**Pull request merged**